code/controllers/subsystems/records.dm

```diff
@@ -179,54 +179,26 @@
 /datum/controller/subsystem/records/proc/reset_manifest()
 	manifest.Cut()
 
-// The one and only method for showing a pop-up crew manifest (browser) window
-/proc/open_crew_manifest(var/mob/user, var/OOC = FALSE)
-	if(!user)
-		return
-	var/const/windowname = "manifest"
-	var/dat = {"<h2 style="text-align: center">Crew Manifest</h2>"}
-	dat += SSrecords.get_manifest(OOC)
-	send_theme_resources(user)
-	user << browse(enable_ui_theme(user, dat), "window=[windowname];size=450x600")
-	return windowname
-
-/datum/controller/subsystem/records/proc/get_manifest(var/OOC = FALSE)
-	var/const/style = {"
-			.manifest {border-collapse: collapse; width: 100%}
-			.manifest td, th {border: 1px solid #DEF; background-color: white; color:black; padding: .25em}
-			.manifest th {height: 2em; background-color: #3F668F; color: white}
-			.manifest tr.head th {background-color: #006E7A;}
-			.manifest td:first-child {text-align: right}
-			.manifest tr.alt td {background-color: #DEF}
-		"}
-	var/dat = {"
-			<head><style>[style]</style></head>
-			<table class="manifest">
-			<tr class='head'><th>Name</th><th>Rank</th><th>Activity</th></tr>
-		"}
-	var/even = 0
-	var/list/isactive = new()
-	for(var/mob/M in player_list)
-		if (OOC)
-			if(M.client && M.client.inactivity <= 10 * 60 * 10)
-				isactive[M.real_name] = "Active"
-			else
-				isactive[M.real_name] = "Inactive"
-		else
-			isactive[M.real_name] = 0
-
-	var/manifest = get_manifest_list()
-	for(var/dep in manifest)
-		var/list/depI = manifest[dep]
-		if(depI.len > 0)
-			dat += "<tr><th colspan=3>[dep]</th></tr>"
-			for(var/list/item in depI)
-				dat += "<tr[even ? " class='alt'" : ""]><td>[item["name"]]</td><td>[item["rank"]]</td><td>[isactive[item["name"]] ? isactive[item["name"]] : item["active"]]</td></tr>"
-				even = !even
-	dat += "</table>"
-	dat = replacetext(dat, "\n", "") // so it can be placed on paper correctly
-	dat = replacetext(dat, "\t", "")
-	return dat
+/datum/controller/subsystem/records/CanUseTopic(var/mob/user, var/datum/topic_state/state = default_state) // this is needed because VueUI closes otherwise
+	if(isnewplayer(user))
+		return STATUS_INTERACTIVE
+	if(isobserver(user))
+		return STATUS_INTERACTIVE
+	if(issilicon(user)) // silicons have the show manifest verb
+		return STATUS_INTERACTIVE
+	return ..()
+
+/datum/controller/subsystem/records/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	. = ..()
+	data = . || data || list()
+
+	VUEUI_SET_CHECK(data["manifest"], SSrecords.get_manifest_list(), ., data)
```

Comment:
`VUEUI_SET_CHECK` does not work for lists, it will always set. There is a separate helper for lists.

Alternatively just don't use a SET_CHECK because they are largely unnecessary.